tz/hash_avx2.go

```diff
@@ -0,0 +1,55 @@
+package tz
+
+import (
+	"hash"
+
+	"github.com/nspcc-dev/tzhash/gf127"
+)
+
+type digest2 struct {
+	x [2]gf127.GF127x2
+}
+
+var _ hash.Hash = new(digest2)
+
+func (d *digest2) Write(data []byte) (n int, err error) {
+	n = len(data)
+	for _, b := range data {
+		mulBitRightx2(&d.x[0], &d.x[1], &minmax[(b>>7)&1])
+		mulBitRightx2(&d.x[0], &d.x[1], &minmax[(b>>6)&1])
+		mulBitRightx2(&d.x[0], &d.x[1], &minmax[(b>>5)&1])
+		mulBitRightx2(&d.x[0], &d.x[1], &minmax[(b>>4)&1])
+		mulBitRightx2(&d.x[0], &d.x[1], &minmax[(b>>3)&1])
+		mulBitRightx2(&d.x[0], &d.x[1], &minmax[(b>>2)&1])
+		mulBitRightx2(&d.x[0], &d.x[1], &minmax[(b>>1)&1])
+		mulBitRightx2(&d.x[0], &d.x[1], &minmax[(b>>0)&1])
+	}
+	return
+}
+
+func (d *digest2) Sum(in []byte) []byte {
+	// Make a copy of d so that caller can keep writing and summing.
+	d0 := *d
+	h := d0.checkSum()
+	return append(in, h[:]...)
```

Comment:
mb 

data = make([]byte, hasSize + len(in))
copy(data, in)
copy(data[len(in):], d.checkSum())
return data

Reply:
It looks longer and less understandable IMO.

Reply:
it's more optimised